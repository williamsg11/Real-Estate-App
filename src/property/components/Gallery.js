/** @flow */

import React, {Component, PropTypes} from 'react';
import {
  Dimensions,
  Image,
  StyleSheet,
  TouchableHighlight,
  View,
} from 'react-native';
import Swiper from 'react-native-swiper';
import Ionicons from 'react-native-vector-icons/Ionicons';

export default class Gallery extends Component {
  static propTypes = {
    style: View.propTypes.style,
    images: PropTypes.array.isRequired,
    setSceneType: PropTypes.func.isRequired,
  };

  renderGallery = image => {
    return (
      <Image source={{uri: image}} style={styles.cover} resizeMode="contain" />
    );
  };

  render() {
    const {images, sceneType, setSceneType} = this.props;
    return (
      <View style={styles.container}>
        {sceneType == 'galleryScene' &&
          <TouchableHighlight
            style={styles.shrinkButton}
            onPress={() => setSceneType('detailScene')}>
            <Ionicons name="ios-close" size={40} color="white" />
          </TouchableHighlight>}
        <Swiper
          style={[styles.gallery]}
          showsButtons
          buttonWrapperStyle={styles.navButtons}>
          {images.map((image, i) => {
            return (
              <View style={styles.album} key={i}>
                {this.renderGallery(image)}
              </View>
            );
          })}
        </Swiper>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flexGrow: 1,
    backgroundColor: '#000',
  },
  album: {
    backgroundColor: '#000',
    width: Dimensions.get('window').width,
    height: Dimensions.get('window').height - 100,
    elevation: 12,
    shadowColor: '#000000',
    shadowOpacity: 0.5,
    shadowRadius: 8,
    shadowOffset: {
      height: 8,
    },
  },
  cover: {
    alignSelf: 'center',
    width: Dimensions.get('window').width,
    height: Dimensions.get('window').height - 100,
  },
  label: {
    margin: 16,
    color: '#fff',
  },
  shrinkButton: {
    position: 'absolute',
    top: 0,
    left: 0,
    padding: 10,
    zIndex: 1000,
  },
  navButtons: {
    backgroundColor: 'transparent',
    flexDirection: 'row',
    position: 'absolute',
    top: 0,
    left: 0,
    flex: 1,
    paddingHorizontal: 10,
    paddingVertical: 10,
    justifyContent: 'space-between',
    alignItems: 'center',
  },
});
