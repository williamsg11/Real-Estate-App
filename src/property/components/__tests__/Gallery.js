import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Gallery from './../Gallery';

test('renders heart', () => {
  const tree = renderer
    .create(
      <Gallery
        images={['http://hello.com/test.png']}
        setSceneType={() => {}}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
