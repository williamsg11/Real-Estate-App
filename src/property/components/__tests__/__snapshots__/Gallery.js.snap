// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders heart 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000",
      "flexGrow": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "position": "relative",
        },
        Object {
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <RCTScrollView
      alwaysBounceHorizontal={true}
      alwaysBounceVertical={false}
      automaticallyAdjustContentInsets={false}
      autoplay={false}
      autoplayDirection={true}
      autoplayTimeout={2.5}
      bounces={false}
      buttonWrapperStyle={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "left": 0,
          "paddingHorizontal": 10,
          "paddingVertical": 10,
          "position": "absolute",
          "top": 0,
        }
      }
      contentContainerStyle={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            undefined,
          ],
        ]
      }
      contentOffset={Object {}}
      horizontal={true}
      index={0}
      loadMinimal={false}
      loadMinimalSize={1}
      loop={true}
      onContentSizeChange={null}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={undefined}
      onResponderTerminationRequest={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onScrollShouldSetResponder={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollsToTop={false}
      sendMomentumEvents={true}
      showsButtons={true}
      showsHorizontalScrollIndicator={false}
      showsPagination={true}
      showsVerticalScrollIndicator={false}
      style={
        Array [
          Object {
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "overflow": "scroll",
          },
          Array [
            undefined,
          ],
        ]
      }
    >
      <RCTScrollContentView
        collapsable={false}
        removeClippedSubviews={true}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                undefined,
              ],
            ],
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 1334,
                "width": 750,
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#000",
                "elevation": 12,
                "height": 1234,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 8,
                },
                "shadowOpacity": 0.5,
                "shadowRadius": 8,
                "width": 750,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "uri": "http://hello.com/test.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 1234,
                  "width": 750,
                }
              }
            />
          </View>
        </View>
      </RCTScrollContentView>
    </RCTScrollView>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "left": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "left": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "position": "absolute",
            "top": 0,
          },
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#007aff",
                "fontFamily": "Arial",
                "fontSize": 50,
              }
            }
          >
            ‹
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#007aff",
                "fontFamily": "Arial",
                "fontSize": 50,
              }
            }
          >
            ›
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
